lib/services/role.go

```diff
@@ -980,6 +980,15 @@ func (set RoleSet) WithoutImplicit() (out RoleSet) {
 	return out
 }
 
+func (set RoleSet) PinSourceIP() bool {
```

Comment:
Add godoc. I find it helpful to document whether this is an "and" or an "or" option.

```
// PinSourceIP determines if the role set should use source IP pinning.
// If one or more roles in the set requires IP pinning then it will be enabled.
```